server/src/main/java/org/spine3/server/entity/storage/ColumnTypeRegistry.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.primitives.Primitives;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * A registry of type conversion strategies for the {@link Column Columns}.
+ *
+ * @param <C> storage-specific implementation of the {@link ColumnType}
+ * @author Dmytro Dashenkov
+ */
+public final class ColumnTypeRegistry<C extends ColumnType> {
+
+    private final ImmutableMap<Class, C> columnTypeMap;
+
+    private ColumnTypeRegistry(Builder<C> builder) {
+        this.columnTypeMap = ImmutableMap.copyOf(builder.columnTypeMap);
+    }
+
+    /**
+     * @return an empty registry, used for
+     * {@linkplain org.spine3.server.storage.memory.InMemoryStorageFactory in-memory storages} etc.
+     */
+    public static <C extends ColumnType> ColumnTypeRegistry<C> empty() {
```

Comment:
The more I look at this method, the more I think it belongs to the `InMemoryStorageFactory` under the `emptyColumnRegistry` name.

It's just a stranger to the public API.